staging/src/k8s.io/kubectl/pkg/util/completion/completion.go

```diff
@@ -167,6 +194,26 @@ func CompGetContainers(f cmdutil.Factory, podName string, toComplete string) []s
 	return CompGetFromTemplate(&template, f, "", []string{"pod", podName}, toComplete)
 }
 
+// CompGetContainersPorts gets the list of containers of the specified pod which begin with `toComplete`.
+func CompGetContainersPorts(f cmdutil.Factory, resourceType string, resourceName string, toComplete string) []string {
+	var template string
+	exposedPort := strings.Split(toComplete, ":")[0]
+
+	if resourceType == "services" || resourceType == "service" || resourceType == "svc" {
```

Comment:
I don't think this if block is ever used.

if I create a deployment w/ port 90, and then a service that exposes port 80 (I intentionally made it different to show what I mean):
```
kubectl apply -f - << EOF
apiVersion: apps/v1
kind: Deployment
metadata:
  name: foo
  labels:
    app: foo
spec:
  replicas: 3
  selector:
    matchLabels:
      app: foo
  template:
    metadata:
      labels:
        app: foo
    spec:
      containers:
      - name: sysinfo 
        image: brianpursley/system-info
        env:
        - name: PORT
          value: "90"
        ports:
        - containerPort: 90
EOF

kubectl apply -f - << EOF
apiVersion: v1
kind: Service
metadata:
  name: foo
  labels:
    app: foo
spec:
  ports:
  - name: http
    port: 80
    protocol: TCP
    targetPort: 8080
  selector:    
    app: foo
  type: NodePort
EOF
```

If I do `kubectl port-forward services/foo <TAB>`, it completes `90:90`, not `80:80`

So it seems to be looking at the underlying pod's containerPort, not at the service's port.

Reply:
I'm wondering if this if statement could just be removed altogether.